### PR TITLE
fix: Replace use of outdated `test` flags and improve quoting

### PR DESCRIPTION
**Pull request by hyperupcall**
This is a fairly straightforward MR that replaces the use of outdated `test` flags and slightly improves quoting.

### CODE REVIEW

File: git-cleanup [hunk 1]
```diff
@@ -36,7 +36,7 @@ while getopts nlrsvh flag; do
 done
 shift $(($OPTIND - 1))
 
-if [ $remotes -eq 0 -a $locals -eq 0 ]; then
+if [ $remotes -eq 0 ] && [ $locals -eq 0 ]; then
```
Comment (nvie) [inline git-cleanup:39]: Why make this change? The `-a` flag works everywhere, in `/bin/sh` and `/bin/bash`. I believe the `&&` approach will only work in systems with `/bin/bash`, so I don't really see the benefit here.

File: git-last-commit-to-file [hunk 2]
```diff
@@ -24,6 +24,4 @@ if [ $# -lt 1 ]; then
     exit 2
 fi
 
-files="$@"
-
-git log -1 --format='%H' -- $files 2>/dev/null
+git log -1 --format='%H' -- "$@" 2>/dev/null
```
Comment (nvie) [inline git-last-commit-to-file:27]: Good change 👍

File: git-local-commits [hunk 3]
```diff
@@ -7,4 +7,4 @@ if ! git remote-branch-exists origin "$curr"; then
     exit 1
 fi
 
-git log $(git merge-base origin/"$curr" "$curr")..."$curr" "$@"
+git log "$(git merge-base origin/"$curr" "$curr")"..."$curr" "$@"
```
Comment (nvie) [inline git-local-commits:10]: Ah yes, good change — thanks 👍

File: git-trash [hunk 4]
```diff
@@ -8,7 +8,7 @@ usage () {
     echo "-h    Show this help" >&2
 }
 
-while getopts sqh flag; do
+while getopts h flag; do
```
Comment (nvie) [inline git-trash:11]: Nice cleanup! Also… I realize `git-trash` isn't even documented.

**Review by nvie — CHANGES_REQUESTED** [review]
Would you mind reverting the `-a` to `&&`, and `-o` to `||` changes? Happy to merge in the rest! Thanks! 👍

File: git-cleanup (hunk 1 above)
Comment (harkabeeparolus) [reply git-cleanup:39]: From personal experience, this is detected by [ShellCheck SC2107](https://www.shellcheck.net/wiki/SC2107), and one of the sources they cite is [BashPitfalls 6](https://mywiki.wooledge.org/BashPitfalls#pf6):

> The binary -a and -o, and ( / ) (grouping) operators are XSI extensions to the POSIX standard. All are marked as obsolescent in POSIX-2008. They should not be used in new code.

Comment (nvie) [reply git-cleanup:39]: Oh, I see. I think I mis-read the original diff and thought it "just" changed the `-a` to `&&` like`if [ $remotes -eq 0 && $locals -eq 0 ]`, as if the `&&` was an argument to `test` (aka `[`). Now I see it just splits the command into two separate `[` commands, with `&&` being normal shell operators to combine them. That's great actually! Thanks for pointing to these resources 👍

**Review by nvie — APPROVED** [review]

**Comment by nvie** [comment]
Thanks for contributing @hyperupcall! 🙏

File: git-cleanup (hunk 1 above)
Comment (hyperupcall) [reply git-cleanup:39]: Thank you @harkabeeparolus for adding the extra detail! Just got back to this today

**Pull request merged**